### PR TITLE
Prerelease checkpoint 3

### PR DESCRIPTION
**Pull request by mullermp**
PR will fix a bunch of other issues

### CODE REVIEW

File: codegen/build.gradle.kts [hunk 1]
```diff
@@ -249,14 +249,12 @@ subprojects {
         tasks["spotbugsTest"].enabled = false
 
         // Configure the bug filter for spotbugs.
-        tasks.withType<com.github.spotbugs.SpotBugsTask> {
-            ignoreFailures = false
-
-            effort = "max"
-            excludeFilterConfig = project.resources.text.fromFile("${project.rootDir}/config/spotbugs/filter.xml")
-            reports {
-                xml.setEnabled(false)
```
Comment (mullermp) [inline codegen/build.gradle.kts:258]: How do we enable html and disable xml reports?

File: codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/ErrorsGeneratorBase.java [hunk 2]
```diff
@@ -35,7 +36,7 @@
 /**
  * ErrorGeneratorBase that protocol generators can extend to get
  * common functionality.
- *
+ * <p>
```
Comment (mullermp) [inline codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/ErrorsGeneratorBase.java:39]: Random <p>

File: codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/ErrorsGeneratorBase.java [hunk 3]
```diff
@@ -162,23 +163,29 @@ public void renderRbsErrorCode() {
...
 
     private void renderServiceModelErrors() {
-        errorShapes.forEach(error -> {
-            String errorName = symbolProvider.toSymbol(error).getName();
-            ErrorTrait errorTrait = error.getTrait(ErrorTrait.class).get();
-            String apiErrorType = getApiErrorType(errorTrait);
-
-            renderServiceModelError(errorName, apiErrorType);
-        });
+        errorShapes
```
Comment (mullermp) [inline codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/ErrorsGeneratorBase.java:166]: Is this refactorable so we don't have to do it twice?

Comment (alextwoods) [reply codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/ErrorsGeneratorBase.java:166]: Yes, but technically changes the interface slightly (set to list).  fine for pre-release here, so I've done that.

**Pull request merged**